resources/views/addproduct.blade.php

```diff
@@ -8,46 +8,46 @@
                     <div class="panel-heading">Add Product</div>
 
                     <div class="panel-body">
-                        <form class="form-horizontal" method="POST" action="/products/index">
+                        <form class="form-horizontal" method="POST" action="/products/edit">
```

Comment:
Your form action is in a route that not exists. This form action should be to a route that is using the store method on your ProductController